django/template/base.py

```diff
@@ -950,11 +956,16 @@ def _resolve_lookup(self, context):
 
 
 class Node:
+    __slots__ = ("token", "origin")
+
     # Set this to True for nodes that must be first in the template (although
     # they can be preceded by text nodes.
     must_be_first = False
     child_nodelists = ("nodelist",)
-    token = None
+
+    def __init__(self):
+        self.token = None
+        self.origin = None
```

Comment:
Last time round, this was `__new__` because subclasses won't necessarily be calling `super().__init__()`
You've introduced the necessary super call for the various bits in _Django_ proper, but won't this still be a problem for userland Node subclasses?

Reply:
Ah yes, thank you for spotting. Yes, I went with the `__init__` approach this time to see how it shakes out. We could still go back to `__new__` if we think it's likely disruptive to enforce calling `super()` in `__init__`. My gut feeling is that not that many custom tags have custom node classes, so it won't be too disruptive, but I didn't do much research there.